src/Microsoft.DotNet.Wpf/src/WindowsBase/System/Windows/Input/KeyConverter.cs

```diff
@@ -4,246 +4,326 @@
+        /// Used to check whether we can convert specified value to <see langword="string"/>.
         /// </summary>
         /// <param name="context">ITypeDescriptorContext</param>
         /// <param name="destinationType">Type to convert to</param>
-        /// <returns>true if conversion is possible</returns>
+        /// <returns><see langword="true"/> if conversion to <see langword="string"/> is possible, <see langword="false"/> otherwise.</returns>
         public override bool CanConvertTo(ITypeDescriptorContext context, Type destinationType)
         {
-            // We can convert to a string.
-            // We can convert to an InstanceDescriptor or to a string.
-            if (destinationType == typeof(string))
-            {
-                // When invoked by the serialization engine we can convert to string only for known type
-                if (context != null && context.Instance != null)
-                {
-                    Key key = (Key)context.Instance;
-                    return ((int)key >= (int)Key.None && (int)key <= (int)Key.DeadCharProcessed);
-                }
-            }
-            return false;
+            // We can convert to a string
+            if (destinationType != typeof(string))
+                return false;
+
+            // When invoked by the serialization engine we can convert to string only for known type
+            if (context is null || context.Instance is null)
+                return false;
+
+            return IsDefinedKey((Key)context.Instance);
         }
 
         /// <summary>
-        /// ConvertFrom()
+        /// Converts <paramref name="source"/> of <see langword="string"/> type to its <see cref="Key"/> represensation.
         /// </summary>
-        /// <param name="context"></param>
-        /// <param name="culture"></param>
-        /// <param name="source"></param>
-        /// <returns></returns>
-        /// <ExternalAPI/> 
+        /// <param name="context">Parser Context</param>
+        /// <param name="culture">Culture Info</param>
+        /// <param name="source">Key String</param>
+        /// <returns>A <see cref="Key"/> representing the <see langword="string"/> specified by <paramref name="source"/>.</returns>
         public override object ConvertFrom(ITypeDescriptorContext context, CultureInfo culture, object source)
         {
-            if (source is string)
-            {
-                string fullName = ((string)source).Trim();
-                object key = GetKey(fullName, CultureInfo.InvariantCulture);
-                if (key != null)
-                {
-                    return ((Key)key);
-                }
-                else
-                {
-                    throw new NotSupportedException(SR.Format(SR.Unsupported_Key, fullName));
-                }
-            }
-            throw GetConvertFromException(source);
+            if (source is not string stringSource)
+                throw GetConvertFromException(source);
```

Comment:
Could you revert this usage of GetConvertFromException and use the previous code `throw new NotSupportedException(SR.Format(SR.Unsupported_Key, fullName));`. This is a change of behavior since it changes the exception message and AFAIK GetConvertFromException returns a non-localized message while the exceptions in WPF are localized. If you want to do the change you should probably do it in a separate PR just to keep this PR a 1:1 replacement.

Reply:
@ThomasGoulet73 Can I resolve the conversation? I believe I've addressed your concern with it.